redaxo/src/core/lib/login/login.php

```diff
@@ -533,10 +533,18 @@ public static function regenerateSessionId()
 
             rex_csrf_token::removeAll();
 
-            rex_extension::registerPoint(new rex_extension_point('SESSION_REGENERATED', null, [
+            $extensionPoint = new rex_extension_point('SESSION_REGENERATED', null, [
                 'previous_id' => $previous,
                 'new_id' => session_id(),
-            ], true));
+                'class' => static::class,
+            ], true);
+
+            // We don't know here if packages have already been loaded
+            // Therefore we call the extension point twice, directly and after PACKAGES_INCLUDED
+            rex_extension::registerPoint($extensionPoint);
+            rex_extension::register('PACKAGES_INCLUDED', static function () use ($extensionPoint) {
+                rex_extension::registerPoint($extensionPoint);
+            }, rex_extension::EARLY);
```

Comment:
Warum willst du mit aller gewalt den EP so früh ablaufen lassen?

wäre es schlimm denn ep sehr spät laufen zu lassen sodass andere zeit haben sich einzuklinken?ä(z.b. request-shutdown)

Reply:
Folgendes Szenario:

```php
// Frontend führt Login durch, dabei wird irgendwann auch das aufgerufen
rex_login::regenerateSessionId();

// noch im gleichen Call passiert irgendwo ein Check, ob es ein Backend-User eingeloggt ist
rex_backend_login:.createUser()?->isAdmin();
// an der stelle wird das Backend dann prüfen, ob die Session noch in der Session-Tabelle steht (also noch gültig ist)
// daher muss das Backend bis dahin bereits die Session-ID gegen die neugenerierte ausgetauscht haben
```

Ich denke ähnliches Szenario mit umgekehrten Rollen von Frontend/Backend-Loginsystem ist auch denkbar.

Reply:
Hab keine idee wie man es besser mache man.

extensions für den EP werden dann im edge case 2x aufgerufen, oder?

Reply:
Es gibt drei Situationen:
1. Addons wurden noch nicht geladen:
    - Dann wird der EP zwei Mal aufgerufen
    - Einmal sofort. Da kann sich aber noch kein Addon drangehangen haben. Da hängt nur unsere Core-Extension dran aus der core-boot.php
    - Und dann nach PACKAGES_INCLUDED noch mal. Da können dann auch Extensions aus Addons dran hängen
    - Doppelt ausgeführt wird also nur die Core-Extension, wobei die in dem Fall sowieso abbricht, da es der eigene Backend-Login ist
2. PACKAGES_INCLUDED war schon:
    - Dann wird der EP nur einmal aufgerufen, denn unser Code für PACKAGES_INCLUDED hier greift ja nicht mehr, da PACKAGES_INCLUDED schon vorbei
3. Eine Situation dazwischen, die Addons sind gerade beim Laden (Booten), und direkt aus der boot.php heraus löst ein Addon die Session-ID-Regenerierung an:
    - Falls ein vorher gebootetes Addon seine Extension schon registriert hat, dann wird diese doppelt ausgeführt

Ich glaube Fall 3 ist eher unwahrscheinlich. Daher finde ich die Lösung zwar auch nicht schön, aber glaube trotzdem ok.

Reply:
Realistisch betrachtet, spielt sich das Ganze sowieso nur zwischen Core und YCom ab, denke ich.
Und für die Kombi sollte meine Variante passen.

Wobei ich dieses Mal mit dem Merge auch warten möchte, bis @dergel es in ycom getestet hat.

Reply:
Ich habe das getestet und soweit klappt es für mich.